helm-chart/renku-core/templates/deployment.yaml

```diff
@@ -1,98 +1,109 @@
+{{- range $version := .Values.versions }}
+---
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: {{ include "renku-core.fullname" . }}
+  name: {{ include "renku-core.fullname" $ }}-{{ $version.name }}
   labels:
-{{ include "renku-core.labels" . | indent 4 }}
+    app.kubernetes.io/deploymentVersion: {{ $version.name }}
+{{ include "renku-core.labels" $ | indent 4 }}
 spec:
-  replicas: {{ .Values.replicaCount }}
+  replicas: {{ $.Values.replicaCount }}
   strategy:
     type: Recreate
   selector:
     matchLabels:
-      app.kubernetes.io/name: {{ include "renku-core.name" . }}
-      app.kubernetes.io/instance: {{ .Release.Name }}
+      app.kubernetes.io/name: {{ include "renku-core.name" $ }}
+      app.kubernetes.io/instance: {{ $.Release.Name }}
+      app.kubernetes.io/deploymentVersion: {{ $version.name }}
   template:
     metadata:
       annotations:
         prometheus.io/scrape: 'true'
         prometheus.io/path: '/metrics'
         prometheus.io/port: '8765'
       labels:
-        app.kubernetes.io/name: {{ include "renku-core.name" . }}
-        app.kubernetes.io/instance: {{ .Release.Name }}
-        {{ include "call-nested" (list . "redis" "redis.fullname") }}-client: "true"
+        app.kubernetes.io/name: {{ include "renku-core.name" $ }}
+        app.kubernetes.io/instance: {{ $.Release.Name }}
+        app.kubernetes.io/deploymentVersion: {{ $version.name }}
+        {{ include "call-nested" (list $ "redis" "redis.fullname") }}-client: "true"
     spec:
-    {{- with .Values.imagePullSecrets }}
+    {{- with $.Values.imagePullSecrets }}
       imagePullSecrets:
         {{- toYaml . | nindent 8 }}
     {{- end }}
       volumes:
         - name: shared-volume
           emptyDir: {}
       containers:
-      {{ if .Values.metrics.enabled }}
-        - name: {{ .Chart.Name}}-rqmetrics
-          image: "{{ .Values.metrics.image.repository }}:{{ .Values.metrics.image.tag }}"
-          imagePullPolicy: {{ .Values.metrics.image.pullPolicy }}
+      {{ if $.Values.metrics.enabled }}
+        - name: {{ $.Chart.Name}}-rqmetrics
+          image: "{{ $.Values.metrics.image.repository }}:{{ $.Values.metrics.image.tag }}"
+          imagePullPolicy: {{ $.Values.metrics.image.pullPolicy }}
           env:
             - name: RQ_REDIS_HOST
-              value: {{ include "call-nested" (list . "redis" "redis.fullname") }}-master
+              value: {{ include "call-nested" (list $ "redis" "redis.fullname") }}-master
       {{ end }}
-        - name: {{ .Chart.Name }}
-          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
-          imagePullPolicy: {{ .Values.image.pullPolicy }}
-          {{ if .Values.debug }}
-          args: ["api", "--debug"]
+        - name: {{ $.Chart.Name }}
+          image: "{{ $version.image.repository }}:{{ $version.image.tag }}"
+          imagePullPolicy: {{ $version.image.pullPolicy }}
           securityContext:
+            {{- if $.Values.debug }}
```

Comment:
Why only if debug is set to true? I'd either always run as user 1000 or add the `runAsNonRoot = true` here as well.

Reply:
👍